plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/matcher/DialogWithTitleMatcher.java

```diff
@@ -10,61 +10,48 @@
  ******************************************************************************/ 
 package org.jboss.reddeer.jface.matcher;
 
-import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.Control;
-import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Shell;
 import org.hamcrest.BaseMatcher;
 import org.hamcrest.Description;
 import org.hamcrest.Matcher;
-import org.jboss.reddeer.core.handler.CompositeHandler;
-import org.jboss.reddeer.core.handler.LabelHandler;
 import org.jboss.reddeer.core.matcher.WithTextMatcher;
+import org.jboss.reddeer.jface.dialogs.TitleAreaDialog;
+import org.jboss.reddeer.swt.impl.shell.DefaultShell;
 
 /**
  * Matcher which matches dialogs title with given title text.
  * @author rawagner
  *
  */
-public class DialogWithTitle extends BaseMatcher<String>{
+public class DialogWithTitleMatcher extends BaseMatcher<String>{
 	
 	private Matcher<String> matcher;
 	
 	/**
 	 * 
 	 * @param title Dialog's title
 	 */
-	public DialogWithTitle(String title) {
+	public DialogWithTitleMatcher(String title) {
 		this(new WithTextMatcher(title));
 	}
 	
 	/**
 	 * 
 	 * @param matcher which matches dialog's title
 	 */
-	public DialogWithTitle(Matcher<String> matcher) {
+	public DialogWithTitleMatcher(Matcher<String> matcher) {
 		this.matcher = matcher;
 	}
 
 	@Override
 	public boolean matches(Object item) {
 		if(item instanceof Shell){
-			Shell shell = (Shell)item;
-			CompositeHandler compositeHandler = CompositeHandler.getInstance();
-			LabelHandler lHandler = LabelHandler.getInstance();
-			try{
-				Composite shellComposite = (Composite)compositeHandler.getChildren(shell)[0];
-				Control[] compositeChildren = compositeHandler.getChildren(shellComposite);
-				Control labelControl = compositeChildren[2];
-				if(labelControl instanceof Label){
-					String labelText = lHandler.getText((Label)labelControl);
-					return matcher.matches(labelText);
-				}
-			}catch (Exception e) {
-				return false;
+			if(new WindowMatcher(org.eclipse.jface.dialogs.TitleAreaDialog.class).matches(item)){
+				DefaultShell shell = new DefaultShell((Shell)item);
+				TitleAreaDialog ta = new TitleAreaDialog(shell);
```

Comment:
what if something goes wrong here with SWT widget, e.g. gets disposed? It fails on exception - is this desired?